bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -111,68 +183,89 @@ private SecurityContext authenticateBearerToken(String token) throws Authenticat
         }
     }
 
-    private SecurityContext authenticateUsernamePassword(String username, String password)
-            throws AuthenticationException {
-        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(username, password);
+    private SecurityContext authenticateBasicAuth(String credentialString) throws AuthenticationException {
+
+        String[] decodedCredentials = new String(Base64.getDecoder().decode(credentialString), StandardCharsets.UTF_8)
+                .split(":");
+        if (decodedCredentials.length != 2) {
+            throw new AuthenticationException("Invalid Basic authentication credential format");
+        }
+
+        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(decodedCredentials[0],
+                decodedCredentials[1]);
         Authentication auth = userRegistry.authenticate(credentials);
         User user = userRegistry.get(auth.getUsername());
         if (user == null) {
             throw new AuthenticationException("User not found in registry");
         }
-        return new UserSecurityContext(user, auth, "Basic");
+
+        UserSecurityContext context = new UserSecurityContext(user, auth, "Basic");
+        return context;
     }
 
     @Override
-    public void filter(ContainerRequestContext requestContext) throws IOException {
-        try {
-            String altTokenHeader = requestContext.getHeaderString(ALT_AUTH_HEADER);
-            if (altTokenHeader != null) {
-                requestContext.setSecurityContext(authenticateBearerToken(altTokenHeader));
-                return;
-            }
+    public void filter(@Nullable ContainerRequestContext requestContext) throws IOException {
+        if (requestContext != null) {
+            String cacheKey = null;
+            try {
+                String altTokenHeader = requestContext.getHeaderString(ALT_AUTH_HEADER);
+                if (altTokenHeader != null) {
+                    cacheKey = getCacheKey(altTokenHeader);
```

Comment:
Imho it's not necessary to hash this token. Should this be replaced with
```
cacheKey = altTokenHeader;
```

Reply:
Why did you move the cache here?
I do not see any need to use the cache for tokens.